frontend/test/lit/test_detensorize.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# RUN: %PYTHON %s | FileCheck %s
+
+# pylint: disable=missing-module-docstring,missing-function-docstring
+
+from catalyst import qjit
+from catalyst.debug import print_compilation_stage
+
+
+# CHECK-LABEL: public @jit_a_plus_b_times_2
+@qjit(keep_intermediate=True)
+def a_plus_b_times_2(a, b):
+    # CHECK: %extracted
+    # CHECK: tensor.extract
+    c = a + b
+    # CHECK: tensor.from_elements
+    return c + c
+
+
+a_plus_b_times_2(1.0, 2.0)
+print_compilation_stage(a_plus_b_times_2, "HLOLoweringPass")
```

Comment:
Does this work?

```suggestion
a_plus_b_times_2.workspace.cleanup()
```

I would also add a little bit more to the tests, specifically here would be the `arith.addf` operation that would normally not appear. And also `CHECK-NOT: linalg.generic`